src/util/fifo.h

```diff
@@ -9,18 +9,16 @@ template <class DataType>
 class FIFO {
   public:
     explicit FIFO(int size)
-            : m_data(NULL) {
+            : m_data{std::vector<DataType>(size)} {
         size = roundUpToPowerOf2(size);
         // If we can't represent the next higher power of 2 then bail.
         if (size < 0) {
             return;
         }
-        m_data = new DataType[size];
         PaUtil_InitializeRingBuffer(
-                &m_ringBuffer, sizeof(DataType), size, m_data);
+                &m_ringBuffer, sizeof(DataType), size, m_data.data());
```

Comment:
the PA_RingBuffer and the vector now have differing sizes which some other code might rely upon.